ai-agent-api-reliability/naive-api-failure/naive.py

```
@@ -0,0 +1,26 @@
+# The key, pasted straight into the code — like every tutorial.
+# (In the video story: this is the key that "aged out" three weeks after launch.)
+GITHUB_TOKEN = "ghp_thisTokenExpiredWeeksAgo0000000000"
+
+def star_repo(owner, repo):
+    req = urllib.request.Request(
+        f"https://api.github.com/user/starred/{owner}/{repo}",
+        method="PUT",
+        headers={"Authorization": f"token {GITHUB_TOKEN}"},
+    )
+    try:
+        urllib.request.urlopen(req)
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)